pkg/steps/template.go

```diff
@@ -728,6 +731,15 @@ func podJobIsFailed(pod *coreapi.Pod) bool {
 	return false
 }
 
+func isPodRunning(pod *coreapi.Pod) bool {
```

Comment:
I am vaguely uneasy about this -- I know that there's some weirdness with Pod state when `InitContainers` are running. Can we get some YAML dumps of Pods from production and write a suite of tests or something?

Reply:
Yeah this is only true if all pods are ready after all initContainers have finished. Do we have cases with long-running initcontainers?

Reply:
I don't know but I surely don't want to find out.

Reply:
(deploying this as is has the risk of being a highly visible regression in behavior that will surely send some angry users to the triage, so I'd prefer to do the safe thing)

Reply:
I increased the timeout to 30 minutes. It doesn't seem there are any metrics around phase transition time, so we have to rely on guessing.